src/machine.rs

```diff
@@ -249,6 +249,37 @@ impl<'m> Machine<'m> {
         Ok(())
     }
 
+    /// Delete the machine.
+    ///
+    /// Deletes the machine, cleaning up all associated resources.
```

Comment:
The docs need to be update to reflect that machine is shutdown before resources are deleted.